snuba/query/validation/functions.py

```diff
@@ -0,0 +1,31 @@
+import logging
+from typing import Sequence
+
+from snuba.query.data_source import DataSource
+from snuba.query.expressions import Expression
+from snuba.query.functions import is_valid_global_function
+from snuba.query.validation import FunctionCallValidator, InvalidFunctionCall
+
+logger = logging.getLogger(__name__)
+
+
+class AllowedFunctionValidator(FunctionCallValidator):
+    """
+    Validates that the function itself is allowed. This does not validate
+    that the function is correctly formed.
+    """
+
+    def __init__(self, enforce: bool = False):
+        self.__enforce = enforce
```

Comment:
I would make this based on a state setting. If you made the setting an int we could even ramp this up incrementally. That way you don't have to deploy to turn this on.